vm-monitor/src/main/java/com/epam/pipeline/vmmonitor/service/CertificateMonitor.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017-2019 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.vmmonitor.service;
+
+import com.epam.pipeline.vmmonitor.model.cert.PkiCertificate;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Service
+@Slf4j
+public class CertificateMonitor {
+
+    private static final String DELIMITER = ",";
+    private final List<String> directoriesToScan;
+    private final List<String> certificateMasks;
+    private final int daysToNotify;
+    private final VMNotificationService notificationService;
+
+    @Autowired
+    public CertificateMonitor(final VMNotificationService notificationService,
+                              @Value("${monitor.cert.scan.folders}") final String directoriesToScan,
+                              @Value("${monitor.cert.file.masks:pem}") final String certificateMasks,
+                              @Value("${monitor.cert.expiration.notification.days:5}") final int daysToNotify) {
+        this.notificationService = notificationService;
+        this.directoriesToScan = Arrays.asList(directoriesToScan.split(DELIMITER));
+        this.certificateMasks = Arrays.asList(certificateMasks.split(DELIMITER));
+        this.daysToNotify = daysToNotify;
+    }
+
+    public void checkCertificates() {
+        directoriesToScan.forEach(this::scanDirForCertificates);
+    }
+
+    private void scanDirForCertificates(final String dirPath) {
+        if (dirPath.isEmpty()) {
```

Comment:
I meant something like StringUtils.isBlank(dirPath) - to prevent listing of empty or null string